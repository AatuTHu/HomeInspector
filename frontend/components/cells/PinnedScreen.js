import variables from '../../env'
import { useState } from 'react'
import { styles } from '../../styles/styles'
import { View, ScrollView,Text } from 'react-native'
import PinnedCard from '../molecules/PinnedCard'
import InfoBox from '../atoms/InfoBox'

export default function PinnedScreen({temperature, setTemperature, humidity, setHumidity}) {

  const { humidityURL, temperatureURL, apiKey} = variables
  
  let noPins = 0

  const onPinPress = async(id, type) => {

    let pinned = false;
    let url = type === 'humidity' ? humidityURL : temperatureURL
    let data = { apiKey, id, pinned }

    const response = await fetch(url, {
      method: "PUT",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(data)
    }).catch((error) => { 
      console.error(error.message); 
    }); 

    if(response.status === 200) {
      type === 'humidity' ?  tempArray = [...humidity] : tempArray = [...temperature]    
      let tempArray
      let objectToBeUpdated = tempArray.findIndex(p => p.id === id);   
      tempArray[objectToBeUpdated].pinned = false;
      type === 'humidity' ? setHumidity(tempArray) : setTemperature(tempArray)
    }

  }

  const renderPinnedCards = (data) => {
    const pinnedItems = data.filter(item => item.pinned === true);

    if (pinnedItems.length === 0) {
      noPins = noPins+1
      return (<></>)
    }

    return pinnedItems.map((item) => (
      <View key={item.id}>
        <PinnedCard
          data={item.temperature || item.humidity}
          location={item.location}
          time={item.time}
          date={item.date}
          unit={item.temperature ? "Â°C" : "%"}
          title={item.temperature ? "temperature" : "humidity"}
          id={item.id}
          onPinPress={onPinPress}
        />
      </View>
    ));
  }


if( temperature.length === 0 || humidity.length === 0) {
  return (<View style = {styles.displayBox}>
    <InfoBox text = "No measurements found" type = "heading" textStyle="subHeadingText"/>
  </View>)
} else {
  return (
  <ScrollView>
     <View style = {styles.displayBox}>
      {renderPinnedCards(temperature)}
      {renderPinnedCards(humidity)}
    </View>
    {noPins === 2 ? (<><View style = {styles.displayBox}>
    <InfoBox text = "Nothing pinned yet" type = "heading" textStyle="subHeadingText"/>
    </View></>) : (<></>)}
  </ScrollView>
  )
}}