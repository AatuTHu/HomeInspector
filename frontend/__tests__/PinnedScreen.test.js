import PinnedScreen from "../components/cells/PinnedScreen"
import { render, fireEvent, waitFor } from '@testing-library/react-native';

describe('PinnedScreen', () => {
    it('should render', () => {
        const screen = render(<PinnedScreen/>);
        expect(screen).toMatchSnapshot();
    })

    it('shows a message if no pinns have not been made', () => {
        const {getByText} = render(<PinnedScreen/>);
        expect(getByText('No pinns made yet')).toBeTruthy()
    })
})