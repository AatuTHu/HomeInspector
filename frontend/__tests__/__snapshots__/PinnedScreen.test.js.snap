// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinnedScreen should render 1`] = `
<View>
  <Text>
    NotesScreen
  </Text>
</View>
`;
